File: src/main/kotlin/com/infrastructure/education/account/models/Credential.kt
package com.infrastructure.education.account.models

import jakarta.persistence.*
import java.io.Serializable

enum class CredentialProvider {
    Self
}

@Embeddable
data class CredentialId(
    @Column
    var credentialId: String,

    @Column
    @Enumerated(EnumType.STRING)
    var provider: CredentialProvider
): Serializable

@Entity
@Table(name = "Credentials")
class Credential(
    @EmbeddedId
    var id: CredentialId,

    @Column
    var credentialKey: String,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "account_id")
    val account: Account,
)

File: BlockInfrastructure-Core/Models/Data/Credential.cs
namespace BlockInfrastructure.Core.Models.Data;

public class Credential : AutomaticAuditSupport
{
    public string CredentialId { get; set; }
    public CredentialProvider CredentialProvider { get; set; }
    public string? CredentialKey { get; set; }

    public string UserId { get; set; }
    public User User { get; set; }
}